packages/plugin-commands-installation/src/recursive.ts

```diff
@@ -500,7 +516,7 @@ export function matchDependencies (
   return matchedDeps
 }
 
-export function createMatcher (params: string[]) {
+export function createMatcher (params: string[], ignoredPackages: string[] = []) {
```

Comment:
there is no need to change this function. You need to pass the ignored packages through params. Just prefix them with `!`.

Reply:
Oh I see! I will fix soon 👍

Reply:
Fixed and force-pushed. [dd6765e](https://github.com/pnpm/pnpm/pull/5408/commits/dd6765e0d3c8f6e5f6b50bca67a66a95d8d09362) thanks!

Reply:
I will try to fix the error happend in a test `!is-positive is not in the npm registry, or you have no permission to fetch it.`

This seems to be happend in `outdate` command

Reply:
In current implementation, `!` ignore pattern does not work with `--latest` option( maybe because `--latest` option add `*` to params?). Is this expected behavior?

Reply:
I don't know
I'll check

Reply:
https://github.com/pnpm/pnpm/pull/5432/files#diff-ee9733bdb390aa333add14ec197cf96707f9e79f94623085da37ebf61aebaa99R51

I think this line causes the issue ( and `—latest` option seems to be not related). For example, with `["!react", "!eslint"]` update params, and `react` gets into ignore pattern so `matchedPatternIndex` should be `-1`. However, `eslint` does not match `react` and override `matchedPatternIndex` after.

Reply:
with dependencies like below:
```
"dependencies": {
    "react": "^16.14.0",
    "react-dom": "^16.14.0"
  }
```

and run `pnpm run update '!react' --latest`, `react-dom` only will be updated and this is expected.

However, run `pnpm run update '!react' '!eslint' --latest`, both of `react` and `react-dom` will be updated and this is unexpected...